kernel/networking/port_manager.c

```diff
@@ -2,18 +2,20 @@
 #include "types.h"
 #include "networking/port_manager.h"
 #include "net/internet_layer/ipv4.h"
+#include "math/rng.h"
 
 static port_entry_t g_port_table[PROTO_COUNT][MAX_PORTS];//tab proto/port
 
-static inline bool port_valid(uint16_t p) {
+static inline bool port_valid(uint32_t p) {
```

Comment:
Changing `p` parameter to `uint32_t` doesn't add any value here if all of the arguments ever passed to it are converted/truncated to 16-bits. If every value passed to this function is already between `0` and `MAX_PORTS` then at worst it wastes some CPU cycles for nothing and at best it gets optimized-out.

I removed it again in the [update compiler warnings clean up branch](https://github.com/differrari/RedactedOS/pull/27/commits/38470d2869d5100f60de7b796ff5679c2aa28594#diff-f4f00397147d0aec61f8969593a70eeb4e861ec5b36fedf266e7057534fe5df7L9). It was disused in the network changes which landed in `master` branch anyway, so kind of moot.